src/util/fields.py

```diff
@@ -510,49 +510,69 @@ def rename_field(cr, model, old, new, update_references=True, domain_adapter=Non
+    if cr.rowcount:
         # If a field with the same name already exists for this model (e.g. due to a custom module),
         # rename it to avoid clashing and warn the customer
         custom_name = new + "_custom"
         rename_field(cr, model, new, custom_name, update_references, domain_adapter=None, skip_inherit=skip_inherit)
-        cr.execute("UPDATE ir_model_fields SET name=%s WHERE model=%s AND name=%s RETURNING id", (new, model, old))
-        [fid] = cr.fetchone() or [None]
         msg = (
             "The field %r from the model %r is now a standard Odoo field, but it already existed in the database "
             "(coming from a non-standard module) and thus has been renamed to %r" % (new, model, custom_name)
         )
         add_to_migration_reports(msg, "Non-standard fields")
 
+    cr.execute("UPDATE ir_model_fields SET name=%s WHERE model=%s AND name=%s RETURNING id", (new, model, old))
+    [fid] = cr.fetchone() or [None]
+
     if fid:
+        # recreate the xmlids
         name = IMD_FIELD_PATTERN % (model.replace(".", "_"), new)
-        # In some cases the same field may be present on ir_model_data with both the double __ and single _ name
-        # version. To avoid conflicts (module, name) on the UPDATE below we keep only the double __ version
+
         cr.execute(
             """
              DELETE FROM ir_model_data
-                   WHERE id IN (SELECT unnest((array_agg(id ORDER BY id))[2:count(id)])
-                                  FROM ir_model_data
-                                 WHERE model = 'ir.model.fields'
-                                   AND res_id = %s
-                                 GROUP BY module)
+                   WHERE model = 'ir.model.fields'
+                     AND res_id = %s
+                     AND module != '__export__'
+               RETURNING module
             """,
             [fid],
         )
-        try:
-            with savepoint(cr):
-                cr.execute("UPDATE ir_model_data SET name=%s WHERE model='ir.model.fields' AND res_id=%s", [name, fid])
-        except psycopg2.IntegrityError:
-            # duplicate key. May happen du to conflict between
-            # some_model.sub_id and some_model_sub.id
-            # (before saas~11.2, where pattern changed)
-            name = "%s_%s" % (name, fid)
-            cr.execute("UPDATE ir_model_data SET name=%s WHERE model='ir.model.fields' AND res_id=%s", [name, fid])
+        if cr.rowcount:
+            modules = {mod for (mod,) in cr.fetchall()}
+
+            inserted = psycopg2.extras.execute_values(
+                cr,
+                """
+                INSERT INTO ir_model_data(module, name, model, res_id, noupdate)
+                     VALUES %s
+                ON CONFLICT DO NOTHING
+                  RETURNING module
+                """,
+                [(module, name, 'ir.model.fields', fid, True) for module in modules],
```

Comment:
Since we only vary `module` maybe an `unnest(%s)` with a list of modules is enough instead of execute_many

Reply:
I like it!
And it resolves the old-psycop2-version-that-dont-have-the-fetch-argument issue[^1]

[^1]: https://upgradeci.odoo.com/upgradeci/run/205248